### DISCUSSION TITLE
understand better IORING_SETUP_COOP_TASKRUN and IORING_SETUP_TASKRUN_FLAG

### DISCUSSION THREAD

**Discussion by romange** [axboe/liburing#617] [Q&A] [answered]
Specifically, is it relevant if a single thread operates on io_uring without SQ_POLL mode enabled?

Should we always set both flags to gain performance benefits? Are there any tradeoffs? 
My use-case - the same thread both submits SQEs and processes completions.

Can I assume that kernels before 5.19 will just silently ignore those flags?

**Comment by axboe** [comment, marked as answer]
> Specifically, is it relevant if a single thread operates on io_uring without SQ_POLL mode enabled?

Yes

> Should we always set both flags to gain performance benefits? Are there any tradeoffs?

Only set TASKRUN_FLAG is you rely on io_uring_cqe_peek() operations to be reliable. This is documented in the man page too.

> Can I assume that kernels before 5.19 will just silently ignore those flags?

No, as with any operation, if you pass in flags the kernel doesn't know about, it'll fail with -EINVAL.
